Unable to build on macOS

I am unable to build the webvault on macOS with node+npm. 
To reproduce, follow the build instruction on macOS.

```
bw_web_builds % make full    
./scripts/checkout_web_vault.sh
Input a git ref (commit hash, branch name, tag name, 'master'): master
Using: 'master' to checkout bitwarden/client.
~/bw_web_builds/web-vault ~/bw_web_builds
Already on 'master'
Your branch is up to date with 'origin/master'.
HEAD is now at 4e4de9812 Bumped web version to 2023.2.0 (#4753)
Your branch is up to date with 'origin/master'.
~/bw_web_builds
~/bw_web_builds/web-vault ~/bw_web_builds
From https://github.com/bitwarden/clients
 * branch                master     -> FETCH_HEAD
Already up to date.
Already on 'master'
Your branch is up to date with 'origin/master'.
~/bw_web_builds
./scripts/patch_web_vault.sh
~/bw_web_builds/web-vault ~/bw_web_builds
Patch file not found, using latest
find: -printf: unknown primary or operator
FAILED: line 15, exit code 1. Press any key to exit...
make: *** [patch-web-vault] Error 1
```

This seems to be caused by the missing `-printf` on `find` provided by macOS(or FreeBSD, since macOS the `find` command is based on FreeBSD one).

--- a/Dockerfile
+++ b/Dockerfile
@@ -53,7 +53,7 @@ WORKDIR /vault/apps/web
 RUN npm run dist:oss:selfhost
 
 RUN printf '{"version":"%s"}' \
-      $(git -c 'versionsort.suffix=-' ls-remote --tags --sort='v:refname' https://github.com/dani-garcia/bw_web_builds.git 'v*' | tail -n1 | sed -E 's#.*?refs/tags/v##') \
+      $(git -c 'versionsort.suffix=-' ls-remote --tags --sort='v:refname' https://github.com/dani-garcia/bw_web_builds.git 'v*' | tail -n1 | grep -Eo '[^\/]*$') \
       > build/vw-version.json
 
 # Delete debugging map files, optional

--- a/scripts/apply_patches.sh
+++ b/scripts/apply_patches.sh
@@ -10,7 +10,7 @@ if [[ -z ${PATCH_NAME} ]]; then
     else
         echo "Patch file not found, using latest"
         # If not, use the latest one
-        PATCH_NAME="$(find ../patches -printf "%f\\n" | sort -V | tail -n1)"
+        PATCH_NAME="$(find ../patches -type f -print0 | xargs -0 basename -a | sort -V | tail -n1)"
     fi
 fi
 

--- a/scripts/build_web_vault.sh
+++ b/scripts/build_web_vault.sh
@@ -1,6 +1,6 @@
 #!/usr/bin/env bash
 set -o pipefail -o errexit
-BASEDIR=$(dirname "$(readlink -f "$0")")
+BASEDIR=$(RL=$(readlink -n "$0"); SP="${RL:-$0}"; dirname "$(cd "$(dirname "${SP}")"; pwd)/$(basename "${SP}")")
 
 # Error handling
 handle_error() {
@@ -30,7 +30,7 @@ npm run dist:oss:selfhost
 
 # Create vw-version.json with the latest tag from the remote repo.
 printf '{"version":"%s"}' \
-      "$(git -c 'versionsort.suffix=-' ls-remote --tags --sort='v:refname' https://github.com/dani-garcia/bw_web_builds.git 'v*' | tail -n1 | sed -E 's#.*?refs/tags/v##')" \
+      "$(git -c 'versionsort.suffix=-' ls-remote --tags --sort='v:refname' https://github.com/dani-garcia/bw_web_builds.git 'v*' | tail -n1 | grep -Eo '[^\/]*$')" \
       > build/vw-version.json
 
 popd

--- a/scripts/checkout_web_vault.sh
+++ b/scripts/checkout_web_vault.sh
@@ -1,6 +1,6 @@
 #!/usr/bin/env bash
 set -o pipefail -o errexit
-BASEDIR=$(dirname "$(readlink -f "$0")")
+BASEDIR=$(RL=$(readlink -n "$0"); SP="${RL:-$0}"; dirname "$(cd "$(dirname "${SP}")"; pwd)/$(basename "${SP}")")
 
 # Error handling
 handle_error() {

--- a/scripts/generate_patch_file.sh
+++ b/scripts/generate_patch_file.sh
@@ -1,6 +1,6 @@
 #!/usr/bin/env bash
 set -o pipefail -o errexit
-BASEDIR=$(dirname "$(readlink -f "$0")")
+BASEDIR=$(RL=$(readlink -n "$0"); SP="${RL:-$0}"; dirname "$(cd "$(dirname "${SP}")"; pwd)/$(basename "${SP}")")
 
 # Error handling
 handle_error() {

--- a/scripts/package_web_vault.sh
+++ b/scripts/package_web_vault.sh
@@ -1,6 +1,6 @@
 #!/usr/bin/env bash
 set -o pipefail -o errexit
-BASEDIR=$(dirname "$(readlink -f "$0")")
+BASEDIR=$(RL=$(readlink -n "$0"); SP="${RL:-$0}"; dirname "$(cd "$(dirname "${SP}")"; pwd)/$(basename "${SP}")")
 
 handle_error() {
     read -n1 -r -p "FAILED: line $1, exit code $2. Press any key to exit..." _

--- a/scripts/patch_web_vault.sh
+++ b/scripts/patch_web_vault.sh
@@ -1,6 +1,6 @@
 #!/usr/bin/env bash
 set -o pipefail -o errexit
-BASEDIR=$(dirname "$(readlink -f "$0")")
+BASEDIR=$(RL=$(readlink -n "$0"); SP="${RL:-$0}"; dirname "$(cd "$(dirname "${SP}")"; pwd)/$(basename "${SP}")")
 
 # Error handling
 handle_error() {

--- a/scripts/tar_web_vault.sh
+++ b/scripts/tar_web_vault.sh
@@ -1,6 +1,6 @@
 #!/usr/bin/env bash
 set -o pipefail -o errexit
-BASEDIR=$(dirname "$(readlink -f "$0")")
+BASEDIR=$(RL=$(readlink -n "$0"); SP="${RL:-$0}"; dirname "$(cd "$(dirname "${SP}")"; pwd)/$(basename "${SP}")")
 
 # Error handling
 handle_error() {
@@ -24,7 +24,7 @@ DATE_FORMAT="${DATE_FORMAT:-%Y-%m-%dT%H:%M:%S%z}"
 # Preserve previous output
 if [[ -f "${OUTPUT_NAME}.tar.gz" ]];
 then
-    DATE_SUFFIX=$(date +"${DATE_FORMAT}" -r "${OUTPUT_NAME}.tar.gz")
+    DATE_SUFFIX=$(date -r "${OUTPUT_NAME}.tar.gz" +"${DATE_FORMAT}")
     mv "${OUTPUT_NAME}.tar.gz" "${OUTPUT_NAME}_${DATE_SUFFIX}.tar.gz"
 fi
 
@@ -36,10 +36,16 @@ fi
 
 mv build web-vault
 # Tar the web-vault
-tar -czvf "${OUTPUT_NAME}.tar.gz" web-vault --owner=0 --group=0
+# Check if we are using bsdtar or gnu-tar, bsdtar does not support --owner/--group
+if [[ "$(tar --version)" =~ .*bsdtar.* ]];
+then
+    tar -czvf "${OUTPUT_NAME}.tar.gz" web-vault
+else
+    tar -czvf "${OUTPUT_NAME}.tar.gz" web-vault --owner=0 --group=0
+fi
 
 # Copy the web-vault
-cp -dpr web-vault "${OUTPUT_NAME}"
+cp -pR web-vault "${OUTPUT_NAME}"
 mv web-vault build
 
 popd

